src/main/java/uk/gov/companieshouse/accounts/association/mapper/AssociationsListUserDaoToDtoMapper.java

```diff
@@ -46,13 +47,15 @@ private Function<Association, Association> enrichWithUserEmailAndDisplayName( fi
     protected void enrichWithItems( Page<uk.gov.companieshouse.accounts.association.models.Association> page, @MappingTarget AssociationsList list, @Context Map<String, String> context ){
         final var userId = context.get( "userId" );
         final var pageContent = page.getContent();
-        final var enrichWithUserEmailAndDisplayName = enrichWithUserEmailAndDisplayName( userId );
 
-        final var items =
-        pageContent.parallelStream()
-                   .map( associationCompanyDaoToDtoMapper::daoToDto )
-                   .map( enrichWithUserEmailAndDisplayName )
-                   .toList();
+        List<Association> items = List.of();
+        if ( !pageContent.isEmpty() ) {
```

Comment:
what about null check.

Reply:
We had this discussion on a call together. I originally implemented null checks, but you had me delete them during the mapper ticket. This is because you analysed the situation and realised that it cannot be null at this point in the flow.

In any case, this change I made here only wraps existing logic from master, into a conditional to optimise the code a bit. It used to call the enrichWithUserEmailAndDisplayName method even when the pageContent was empty which was redundant. Now it only does it if the pageContent is not empty.